src/NServiceBus.Core/Transports/Learning/DirectoryBasedTransaction.cs

```diff
@@ -104,10 +104,25 @@ void RecoverPending()
             //only need to move the incoming file
             foreach (var file in pendingDir.EnumerateFiles(TxtFileExtension))
             {
-                File.Move(file.FullName, Path.Combine(basePath, file.Name));
+                var destFileName = Path.Combine(basePath, file.Name);
+                try
+                {
+                    File.Move(file.FullName, destFileName);
```

Comment:
Should we use the AsyncFile.Move instead to get retries here? 

https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/Transports/Learning/AsyncFile.cs#L112

Reply:
I considered this and had an initial version that used it but then I realized that probably ignoring the exception and moving on under the assumption another endpoint takes care of it is the safer approach. Happy to be convinced otherwise

Reply:
I'm ok with that but note that AsyncFile.Move just does a few retries and then returns false ie it would have the same effect but with retries added?

Reply:
going down the road of AsyncFile is much heavier in terms of changes, 
it also introduces async in a path that is not designed to handle it